src/dev_server.rs

```diff
@@ -4,31 +4,73 @@ use clap::Parser;
 use std::{
     fs,
     io::{prelude::*, BufReader},
-    net::{IpAddr, SocketAddr, TcpListener, TcpStream},
+    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener, TcpStream},
     path::Path,
     process,
 };
 
+/// A simple HTTP server that can, optionally, watch a given command.
+///
+/// Get the files at `watch_path` and serve them at a given IP address
+/// (127.0.0.1:8000 by default). An optional command can be run along the
+/// server, relaunched if changes are detected in your project via a [`Watch`].
+///
+/// # Usage
+///
+/// ```no_run
+/// use xtask_wasm::{metadata, DevServer};
+/// use std::process::Command;
+///
+/// let build_path = metadata().workspace_root.join("build");
+/// let mut command = Command::new("cargo");
+/// command.arg("run");
+///
+/// DevServer::new()
+/// .command(command)
+/// .start(build_path)
+/// .expect("cannot run dev server process");
+/// ```
+///
+/// This starts to serve files in the `build` directory, run the `cargo run`
+/// command. If changes are detected in the workspace root (recursively
+/// expect for target directory or hidden files), the `cargo run` command is
+/// killed and re-launched.
 #[non_exhaustive]
 #[derive(Debug, Parser)]
 pub struct DevServer {
+    /// Ip address to bind. Default to `127.0.0.1`
     #[clap(long, default_value = "127.0.0.1")]
     pub ip: IpAddr,
+    /// Port number. Default to `8000`
     #[clap(long, default_value = "8000")]
     pub port: u16,
 
-    #[clap(flatten)]
-    pub watch: Watch,
+    /// Watched command running along the server
     #[clap(skip)]
     pub command: Option<process::Command>,
+    /// Watching process of the server
+    #[clap(flatten)]
+    pub watch: Watch,
 }
 
 impl DevServer {
+    /// Create a new `DevServer`.
+    pub fn new() -> Self {
+        Self {
+            ip: IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+            port: 8_000,
```

Comment:
:information_source: This approach isn't great because you're defining the default values in 2 different places: here in the `new()` constructor and above in the clap's `default_value`.

Again you need to think of the long term. In one month someone is going to be unhappy with the default part and change this one to 3000 and the other will stay on 8000 because they missed it.